google-cloud-tools-plugin/src/com/google/gct/idea/debugger/ui/CloudAttachDialog.java

```diff
@@ -273,7 +273,7 @@ else if (mySyncResult.needsSync()) {
       myWarningLabel2.setVisible(false);
       myInfoPanel.setVisible(true);
     }
-    else if (mySyncResult.needsSync() && mySyncResult.getTargetSyncSHA() == null) {
+    else if (mySyncResult.getTargetSyncSHA() == null) {
```

Comment:
Check out the `else if` clause right about the one I modified:

`else if (mySyncResult.needsSync())`

If needsSync is true, the clause I modified would never have been reached in the first place.

That aside, from debugging through the execution I noticed that for the module with the missing local repo, the SHA was null and needsSync was false causing no warning to be shown.  Looking at ProjectRepositoryValidator.checkSyncStashState() needsSync will always be false if the SHA is null.